### PR TITLE
Migrate octopus to go modules

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Migrate project to `go-mod`
- Bump `controller-runtime` to 0.2.2

**Related issue(s)**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,7 +36,9 @@ deploy: manifests
 # Generate manifests e.g. CRD, RBAC etc.
 .PHONY: manifests
 manifests:
```
Comment (mszostok) [inline Makefile]: adjust that for kubebuilder v2: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v2/Makefile

**Review by klaudiagrz — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (polskikiel) [reply Makefile]: I will stay with the kubebuilder v1 :(

File: pkg/fetcher/pod_test.go [hunk 2]
```diff
@@ -40,32 +39,27 @@ func TestGetPodsForSuite(t *testing.T) {
...
+	sut := fetcher.NewForTestingPod(cli)
 	// WHEN
 	actualPods, err := sut.GetPodsForSuite(context.TODO(), givenSuite)
 	// THEN
 	require.NoError(t, err)
 	require.Len(t, actualPods, 1)
 	assert.Equal(t, givenPod, actualPods[0])
 
+	// WHEN
+	actualPods, err = sut.GetPodsForSuite(context.TODO(), v1alpha1.ClusterTestSuite{ObjectMeta: v12.ObjectMeta{
```
Comment (mszostok) [inline pkg/fetcher/pod_test.go]:
```suggestion
	actualPods, err = sut.GetPodsForSuite(context.Background(), v1alpha1.ClusterTestSuite{ObjectMeta: v12.ObjectMeta{
```

File: hack/mocks-gen.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+for d in ./pkg/* ; do
```
Comment (mszostok) [inline hack/mocks-gen.sh]: generate only required interfaces

**Review by mszostok — APPROVED** [review]
Accepted for now. Needs polishing in future

**Pull request merged**